airflow-core/src/airflow/api_fastapi/core_api/routes/public/dag_parsing.py

```diff
@@ -66,3 +71,63 @@ def reparse_dag_file(
 
     parsing_request = DagPriorityParsingRequest(fileloc=path)
     session.add(parsing_request)
+
+
+@dag_parsing_router.post(
+    "/manage/reserialize",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_404_NOT_FOUND,
+            status.HTTP_409_CONFLICT,
+            status.HTTP_500_INTERNAL_SERVER_ERROR,
+        ]
+    ),
+    dependencies=[Depends(action_logging())],
+)
+def reserialize_dags(
+    request: DagReserializePostBody,
+    session: SessionDep,  # Add your session dependency
+):
+    """
+    Reserialize DAG bundles in Airflow.
+
+    - **bundle_names**: List of specific bundles to reserialize (all if empty)
+    """
+    try:
+        manager = DagBundlesManager()
+
+        # Getting all bundle names which was retrieved in validation function
+        manager.sync_bundles_to_db()
+        all_bundle_names = set(manager.get_all_bundle_names())
+
+        # Validate bundle names if specified
+        if request.bundle_names:
+            bundles_to_process = set(request.bundle_names)
+            if len(bundles_to_process - all_bundle_names) > 0:
+                raise HTTPException(
+                    status.HTTP_400_BAD_REQUEST,
+                    f"Invalid bundle name: {bundles_to_process - all_bundle_names}",
+                )
+        else:
+            bundles_to_process = all_bundle_names
+
+        file_locations = session.scalars(
+            select(DagModel.fileloc).where(DagModel.bundle_name.in_(list(bundles_to_process)))
+        )
+        # Process each bundle
+        parsing_requests = [DagPriorityParsingRequest(fileloc=fileloc) for fileloc in file_locations]
```

Comment:
I think we need to handle the case where there is no bundle configures in the installation. (bundle_name is None), all_bundle_names is empty. And we need to reparse all I guess.